src/inputs/dropdown/query-search-dropdown/PQuerySearchDropdown.vue

```diff
@@ -0,0 +1,255 @@
+import type { SearchDropdownMenuItem } from '@/inputs/dropdown/search-dropdown/type';
+import type { KeyMenuItem, ValueMenuItem, QueryItem } from '@/inputs/search/query-search/type';
+import PSearch from '@/inputs/search/search/PSearch.vue';
+
+export default defineComponent({
+    name: 'PQuerySearchDropdown',
+    components: {
+        PContextMenu,
+        PSearch,
+        PTag,
+        PI,
+    },
+    directives: {
+        focus: vFocus,
+        clickOutside: vOnClickOutside as DirectiveFunction,
+    },
+    model: {
+        prop: 'value',
+        event: 'update:value',
+    },
+    props: {
+        value: {
+            type: String,
+            default: '',
+        },
+        placeholder: {
+            type: String,
+            default: undefined,
+        },
+        focused: {
+            type: Boolean,
+            default: false,
+        },
+        keyItemSets: {
+            // FIXME:: below any type
+            type: Array as PropType<any>,
+            default: () => [],
+        },
+        valueHandlerMap: {
+            type: Object,
+            default: () => ({}),
+        },
+        selected: {
+            type: Array,
+            default: () => [],
+        },
+        multiSelectable: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    setup(props, { emit }: SetupContext) {
+        const {
+            state: querySearchState,
+            focus, blur, hideMenu, showMenu,
+            onInput,
+            onKeydownCheck,
+            onKeyupEnter,
+            onPaste,
+            onDeleteAll,
+            preTreatSelectedMenuItem,
+        } = useQuerySearch(props, { strict: true });
+
+        const state = reactive({
+            proxySelected: useProxyValue('selected', props, emit),
+        });
+
+        /* util */
+        const selectItem = (queryItem: QueryItem) => {
+            if (props.multiSelectable) {
+                state.proxySelected = [...state.proxySelected, queryItem];
+            } else {
+                state.proxySelected = [queryItem];
+            }
+        };
+
+        /* event */
+        const onDeleteTag = (item: SearchDropdownMenuItem, index: number) => {
+            state.proxySelected.splice(index, 1);
+            state.proxySelected = [...state.proxySelected];
+        };
+
+        const onMenuSelect = async (item: KeyMenuItem | ValueMenuItem) => {
+            const queryItem = await preTreatSelectedMenuItem(item);
+            if (queryItem) selectItem(queryItem);
+        };
+
+        const onEnter = async () => {
+            const queryItem = await onKeyupEnter();
+            if (queryItem) selectItem(queryItem);
+        };
+
+        const handleClickDropdownButton = () => {
+            if (querySearchState.visibleMenu) hideMenu();
+            else showMenu(true);
+        };
+
+        return {
+            ...toRefs(querySearchState),
+            ...toRefs(state),
```

Comment:
It's hard to find variables in `state`, I recommend not to use `toRefs` in this case.